### PR TITLE
Improved Menu system.

### PR DESCRIPTION
**Pull request by Dragon20C**
- Redone ui
- Added credits menu
- Added a github button that links to the Org
- Added a global script called GameManager

GameManager is used to load into specific scenes, e.g space station, ship editor, etc

### CODE REVIEW

**Comment by ChaoticTechSupport** [comment]
okay, I looked through it, and here are some proposed changes.

1. could you make a new branch (from main) and copy and paste the code their? (this is due to some of my previous incompetence where you lightmanager branch will overwrite the movement system (will fix when light manager has been pull requested)

2. could you have the MenuBase script extend UserInterface?

3. could you fix your anchors? (set current)

4. smoothie will be angry if I didn't say this, Please add documentation (##). (if you want to check your documentation search your script name in the script search help function)

5. good luck!

**Comment by ChaoticTechSupport** [comment]
> okay, I looked through it, and here are some proposed changes.
> 
> 1. could you make a new branch (from main) and copy and paste the code their? (this is due to some of my previous incompetence where you lightmanager branch will overwrite the movement system (will fix when light manager has been pull requested)
> 2. could you have the MenuBase script extend UserInterface?
> 3. could you fix your anchors? (set current)
> 4. smoothie will be angry if I didn't say this, Please add documentation (##). (if you want to check your documentation search your script name in the script search help function)
> 5. good luck!

(when making a new branch please only add the menu files to it.)

**Comment by ChaoticTechSupport** [comment]
please do not pay attention to my previous Errors. this actually looks Great! but has a couple problems

1. in your main menu tscn you have a NinePatchRect that is a parent of a vbox container, please use a panel container (change the style texture <theme overrides> to your settings on the NinePatchRect), this is due to NinePatchRects not having Container Functionality. (same thing for settings and credits NinePatchRects).
2. in your main menu tscn your label is offset using pixels, and not anchors, this causes problems on larger moniters, please set your anchors to where your label is (or use set current).
3. i could be wrong but i belive it would be possible (and better) for you to commit to main instead of the Menu System branch, as it was not branched off main.
4. while I am thankful that you added how to use the menu, when i said documentation i meant this: https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_documentation_comments.html
5. OPTIONAL: rather then using multiple RitchTextLabels in credits, would it be possible to only use one?
6. could you move the menu scenes into the scenes folder, and the menu scripts into the scripts folder? i think Smoothie wanted us to use a object oriented file system (could be wrong).
7. OPTIONAL: could you orgonize your script using:
8.# --- Class ---

# --- Exported Properties ---

# --- Public Properties ---

# --- Private Properties ---

# --- Enum ---

# --- Signals ---

# --- Built-in Callbacks ---

# --- Custom Methods ---

**Comment by Dragon20C** [comment]
@ChaoticTechSupport  I have updated the project again, Hopefully this time I have followed your advice, also I am not sure how to do the documentation so if something is wrong please guide me, thank you!

**Comment by ChaoticTechSupport** [comment]
> @ChaoticTechSupport I have updated the project again, Hopefully this time I have followed your advice, also I am not sure how to do the documentation so if something is wrong please guide me, thank you!

looks Great! and ready to merge, but for future refrence here are some things to consider:

1. for url buttons there is a built in link button you can use for increased/streamlined preformence.
2. when using text in buttons with textures, it is generally recomemended to use a default button with a texture style box.
3. when documenting A: you dont need to document private functions (just do a basic comment if its a complicated function). B: to create a truly private function, you can prefix it with _
4. when documenting variables, it is usually perfered to have the documentation inline. ex:
variable = 9 ## this variable equals 9.

i think it is ready to merge, but will wait for smoothie to review.

File: Scripts/Menu System/MenuSystem.gd
Comment (StrawberrySmoothieDev) [inline Scripts/Menu System/MenuSystem.gd:36]: This is... odd. What I'm seeing here is that you're preloading all the scenes in advance. While this does have it's merits, it will cause issues once we start adding more to our scenes. If you could explain what your approach is here that would be amazing!

File: Scripts/Player/Player.gd
Comment (StrawberrySmoothieDev) [inline Scripts/Player/Player.gd:3]: This is a menu PR, not sure why we're editing the player. The edits are good, just would like them moved to another PR.

File: Scripts/Templates/InteractableObject/InteractableObjectTemplate.gd
Comment (StrawberrySmoothieDev) [inline Scripts/Templates/InteractableObject/InteractableObjectTemplate.gd:4]: https://docs.godotengine.org/en/stable/tutorials/scripting/creating_script_templates.html

**Review by StrawberrySmoothieDev — CHANGES_REQUESTED** [review]
This PR is pretty good, my only issue is there's a lot of non-menu related stuff here that should probably be in other commits. If you can please try and separate them. Thanks and well done! :3

File: Scripts/InteractionSystem/Base/InteractableObj.gd
Comment (Dragon20C) [reply Scripts/InteractionSystem/Base/InteractableObj.gd:36]: anything related to the interactable system shouldnt be in this project, I will remove it as @ChaoticTechSupport has already created a better system.

Comment (Dragon20C) [reply Scripts/InteractionSystem/Base/InteractableObj.gd:36]: Actually I think this is already in the main repo, so this is not me.

File: Scripts/InteractionSystem/Base/InteractableObj.gd
Comment (Dragon20C) [reply Scripts/InteractionSystem/Base/InteractableObj.gd:22]: Understandable, in future projects I will follow your guidelines!

File: Scripts/InteractionSystem/Base/InteractableObj.gd
Comment (Dragon20C) [reply Scripts/InteractionSystem/Base/InteractableObj.gd:9]: Everything related to interactions is already in the main repo, meaning I am not adding it.

File: Scripts/Player/Player.gd
Comment (Dragon20C) [reply Scripts/Player/Player.gd:3]: already in the main repo.

File: Scripts/Menu System/MenuSystem.gd
Comment (Dragon20C) [reply Scripts/Menu System/MenuSystem.gd:36]: Im not sure what issues could arise from this but its simply like this, load all the menus into a dictionary and then I can instance each menu into a canvas layer node, and its simply queue freeing previous menu and instancing the new menu, its simple as that, also each menu tells the menu system which menu to load next while also keeping track of previous menus so we can go back without the menu needing to know other menus.

File: Scripts/InteractionSystem/Base/InteractableObj.gd
Comment (Dragon20C) [reply Scripts/InteractionSystem/Base/InteractableObj.gd:22]: that makes sense and if my code has private properties I will change it.

File: Scripts/InteractionSystem/Base/InteractableObj.gd
Comment (ChaoticTechSupport) [reply Scripts/InteractionSystem/Base/InteractableObj.gd]: sorry, i forgot to update this to merge with main, could you check again?

File: Scripts/InteractionSystem/Base/InteractableObj.gd
Comment (ChaoticTechSupport) [reply Scripts/InteractionSystem/Base/InteractableObj.gd]: sorry, i forgot to update this to merge with main, could you check again?

File: Scripts/Player/Player.gd
Comment (ChaoticTechSupport) [reply Scripts/Player/Player.gd]: sorry, i forgot to update this to merge with main, could you check again?

**Comment by StrawberrySmoothieDev** [comment]
Just testing checkouts.

File: Scripts/Menu System/MenuSystem.gd
Comment (StrawberrySmoothieDev) [reply Scripts/Menu System/MenuSystem.gd:36]: It's certainly an option, and will be useful when we have a lot more menus, but for now dynamic loading/unloading of menus could cause issues. It may be a better idea to, while we don't have a large amount of menus to keep track of, have them all exist at once in a TabContainer or otherwise. I leave it up to you but it feels unnecessary.

**Comment by StrawberrySmoothieDev** [comment]
Looks good to me, I'm going to run some unit tests then I'll approve it. Thanks!

**Comment by StrawberrySmoothieDev** [comment]
Looks great to me! Love the CRT shader! :3

**Review by StrawberrySmoothieDev — APPROVED** [review]

**Pull request merged**
